aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -156,31 +179,39 @@ SparseTensor new_with_tensor_and_size_sparse(const LongTensor& indices, const Te
 
   int64_t sparseDims = indices.size(0);
   int64_t denseDims = values.dim() - 1;
-  AT_CHECK(sizes.size() == sparseDims + denseDims, "number of dimensions must be sparseDims (", sparseDims, ") + denseDims (", denseDims, "), but got ", sizes);
+  AT_CHECK(sizes.size() == sparseDims + denseDims, "number of dimensions must be sparseDims (", sparseDims, ") + denseDims (", denseDims, "), but got ", sizes.size());
+
+  // Check to make sure all indices are within the boundaries of `sizes`
+  // NOTE: We don't need to check this for scalar sparse tensors, which
+  // has sizes.size() == sparseDims == denseDims == 0
+  if (sizes.size() > 0) {
```

Comment:
shouldn't this condition be  `!indices->is_empty()`?  Otherwise I'm not sure what happens if sparseDims == 0 or if nnz == 0.